### PR TITLE
Implement Initial UI components for discussions

### PR DESCRIPTION
**Pull request by atstoyanov**
For testing purposes line 20 in DiscussionsRoute can be commented, until the feature `communications-discussions` is implemented in the backend.

### CODE REVIEW

File: src/components/composite/common/Discussion/PostComment.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      enableReinitialize
+      onSubmit={handleSubmit}
+    >
+      {({ isValid, dirty }) => (
+        <Form noValidate>
+          <Grid container spacing={2}>
+            <Grid item xs={12}>
+              <FormikMarkdownField name="post" title="Comment on this discussion" placeholder="Comment" />
```
Comment (hero101) [inline src/components/composite/common/Discussion/PostComment.tsx:44]: translation

File: src/components/composite/entities/Communication/DiscussionOverview.tsx [hunk 2]
```diff
@@ -1,8 +1,78 @@
...
+      />
+      <ListItemAvatar>
+        <AvatarGroup
+          max={3}
+          classes={{
+            avatar: styles.avatar,
+          }}
+        >
+          {avatars.map((a, i) => (
+            <Avatar key={i} src="">
```
Comment (hero101) [inline src/components/composite/entities/Communication/DiscussionOverview.tsx:69]: source is missing. Is this a placeholder?

File: src/components/composite/layout/Discussions/DiscussionsLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <Grid item container>
+        <Paper elevation={0} square className={styles.paper}>
+          <Grid container alignItems="center" justifyContent="space-between">
+            <Grid item>
+              <Typography variant="h1">{title}</Typography>
+            </Grid>
+            {allowCreation && (
+              <Grid item>
+                <Link component={RouterLink} to="/discussions/new">
+                  <Button text={'Initiate discussions'} variant="primary" />
```
Comment (hero101) [inline src/components/composite/layout/Discussions/DiscussionsLayout.tsx:38]: translation

File: src/pages/Discussions/DiscussionListPage.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { PageProps } from '../common';
+
+interface DiscussionsPageProps extends PageProps {}
+
+export const DiscussionListPage: FC<DiscussionsPageProps> = ({ paths }) => {
+  useUpdateNavigation({ currentPaths: paths });
+
+  return (
+    <ThemeProviderV2>
+      <DiscussionsLayout title="Discussions" allowCreation>
```
Comment (hero101) [inline src/pages/Discussions/DiscussionListPage.tsx:15]: translation

File: src/routing/discussions/DiscussionsRoute.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import DiscussionPage from '../../pages/Discussions/DiscussionPage';
+import NewDiscussionPage from '../../pages/Discussions/NewDiscussionPage';
+import { nameOfUrl } from '../url-params';
+
+interface DiscussionsRouteProps extends PageProps {}
+
+export const DiscussionsRoute: FC<DiscussionsRouteProps> = ({ paths }) => {
+  const { url, path } = useRouteMatch();
+
+  const currentPaths = useMemo(() => [...paths, { value: url, name: 'Discussions', real: true }], [paths]);
```
Comment (hero101) [inline src/routing/discussions/DiscussionsRoute.tsx:16]: translation

File: src/views/Discussions/DiscussionView.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+        <Grid item>
+          <Filter data={comments}>
+            {filteredComments => {
+              if (filteredComments.length === 0) return null;
+              return (
+                <Box marginTop={2}>
+                  {filteredComments.map((c, i) => (
+                    <>
+                      <Divider />
```
Comment (hero101) [inline src/views/Discussions/DiscussionView.tsx:70]: the first comment will have a divider also; is that on purpose?

File: src/views/Discussions/DiscussionsListView.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      description:
+        'Discussion information to be placed here. In approximately two to three sentences, this is an even long sentence. Another one been added!',
+      date: new Date('2021-10-19'),
+      avatars: ['A', 'B', 'C', 'D'],
+      count: 43,
+    },
+  ];
+
+  return (
+    <ProfileCard title="Discussions" helpText="List of all discussions">
```
Comment (hero101) [inline src/views/Discussions/DiscussionsListView.tsx:48]: translation

**Review by hero101 — CHANGES_REQUESTED** [review]
Also add some `aria-label` attributes, where you find appropriate, for future UI testing

File: src/components/composite/common/Discussion/Comment.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  createdOn: Date;
+  createdBy: string;
+  depth: number;
+}
+
+export const DiscussionComment: FC<DiscussionCommentProps> = ({ commentId, message, createdOn, createdBy, depth }) => {
+  const styles = useStyles();
+  const { t } = useTranslation();
+
+  const depthPadding = depth > 0 ? depth * 8 + 2 : 0;
```
[the remark this answers is not in the document]
Comment (atstoyanov) [reply src/components/composite/common/Discussion/Comment.tsx:27]: It is related to the avatar width, so can be extracted into a constant.

File: src/components/composite/entities/Communication/DiscussionOverview.tsx (hunk 2 above)
Comment (atstoyanov) [reply src/components/composite/entities/Communication/DiscussionOverview.tsx:69]: more like placeholder, but can be added to the model.

File: src/pages/Discussions/DiscussionListPage.tsx (hunk 4 above)
Comment (atstoyanov) [reply src/pages/Discussions/DiscussionListPage.tsx:15]: This should come from somewhere, still unknow.

File: src/views/Discussions/DiscussionsListView.tsx (hunk 7 above)
Comment (hero101) [reply src/views/Discussions/DiscussionsListView.tsx:66]: this is not resolved

File: src/views/Discussions/DiscussionView.tsx (hunk 6 above)
Comment (hero101) [reply src/views/Discussions/DiscussionView.tsx]: Now the last item will have a divider. Typically the items between the first and last should have a divider, or is this fitting the style?

**Review by hero101 — APPROVED** [review]

**Pull request merged**